### PR TITLE
Whack-A-Mole

### PR DESCRIPTION
**Pull request by thoys**
## Whack-a-Mole game for Marketplace

Whack moles with your friends, using the red and blue mallet.

Internal tracker: https://highfidelity.fogbugz.com/f/cases/859/Whack-a-mole

## QA Test

Try testing this with two persons.

1. Load up the whackAMole script to create it. Edit -> Running Scripts -> From URL ( https://cdn.rawgit.com/thoys/hifi/8c7e3fa0ab141da8941612cfa2e3b921244a2795/unpublishedScripts/marketplace/whackAMole/createWhackAMoleConsole.js )
2. Far trigger the push button.
3. Drop the coin into the slot.
4. The game should start. You have 30 seconds to whack as many moles you can. Each player has their own mallet. (left = blue, right = red).
5. The game should move back to attract mode after the game. There is 20% chance that the mole laughs after emerging from its hole.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 8c7e3fa0ab141da8941612cfa2e3b921244a2795 available at: https://gist.github.com/0d68180f9b88bab25f9ce909c9adda33<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10509-8c7e3fa.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10509-8c7e3fa.dmg

**Comment by Nex-Pro** [comment]
Testing this with two HMDs

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/41/HighFidelity-Beta-PR10509-8c7e3fa0ab141da8941612cfa2e3b921244a2795.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/41/HighFidelity-Beta-PR10509-8c7e3fa0ab141da8941612cfa2e3b921244a2795.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/56/HighFidelity-Beta-PR10509-137e7d2d9a7edef090d4ccdac87c384d9df1c8d2.dmg

File: unpublishedScripts/marketplace/whackAMole/createWhackAMoleConsole.js [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+            y: -1.52587890625e-05,
+            z: -1.52587890625e-05
+        },
+        type: 'Model'
+    };
+    return helpers.createEntity(entityProperties, wamConsole);
+}
+
+function createDigitalReadout(wamConsole, id) {
+    var left = id === 0 || id === 1;
```
Comment (huffman) [inline unpublishedScripts/marketplace/whackAMole/createWhackAMoleConsole.js]: Can we label these ids by adding constants somewhere?

File: unpublishedScripts/marketplace/whackAMole/createWhackAMoleConsole.js [hunk 2]
```diff
@@ -0,0 +1,314 @@
...
+            url: model
+        },
+        
+        modelURL: model,
+        name: 'wam-readout_' + (id + 1),
+       
+        type: 'Model'
+    };
+    switch (id) {
+        case 0:
```
Comment (huffman) [inline unpublishedScripts/marketplace/whackAMole/createWhackAMoleConsole.js]: And use them down here

File: unpublishedScripts/marketplace/whackAMole/mole.js [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+    hit: function() {
+        var currentMilliseconds = Date.now();
+        helpers.debugPrint('HIT!!');
+        this.reactAngry();
+        this._playSound(SFX_HIT);
+        // TODO: make sure the dropDown has a high speed (FPS) / HIT_FPS
+        // if (this.state === MOLE_STATE.POPUP) {
+            // TODO: play current frame to down
+
+        // } else if (this.state === MOLE_STATE.UP) {
```
Comment (huffman) [inline unpublishedScripts/marketplace/whackAMole/mole.js]: Should this commented code be removed?

**Comment by Nex-Pro** [comment]
Tested with two people myself and XaosPrincess.

This works pretty well. 

I felt that perhaps there should be a countdown after the coin insertion as there didn't seem to be much time to get the whackers in hand before the game started.
Also Might be nice to have some kind of game over jingle rather than the machine simply stopping.
Xaos suggested that it might be cool for the machine to announce  "Blue Wins"  or "Red Wins" in some dramatic manner.

Because the script to rez this is client side I restarted interface and had the game appear in the ante-room of the tutorial.

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/63/HighFidelity-Beta-PR10509-6155a267554cc7f4cc7f5bd8888a2f4e5f5db60f.exe

**Comment by Nex-Pro** [comment]
### PASS

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/63/HighFidelity-Beta-PR10509-6155a267554cc7f4cc7f5bd8888a2f4e5f5db60f.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/68/HighFidelity-Beta-PR10509-57789210cc7c1be2052a36e698d0862cc7a287a6.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/68/HighFidelity-Beta-PR10509-57789210cc7c1be2052a36e698d0862cc7a287a6.dmg

**Comment by ZappoMan** [comment]
@thoys - why is this PR "Do not merge"?

**Comment by thoys** [comment]
Some changes were requested. Mallets should be clone able instead of the
tractor action and should be the way to start the game and the console will
get a better look.

**Comment by ZappoMan** [comment]
Removed Approvals until @thoys implements additional features requested

**Comment by ZappoMan** [comment]
This PR has been set to Do Not Merge for more than a month. Closing until developer comes up with a plan to get it in shape.

**Pull request closed without merging**